src/repl/App.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { StrictMode } from 'react';
+import ReactDOM from 'react-dom/client';
+import Terminal from './components/Terminal';
+import Editor from './components/Editor';
+import Plot from './components/Plot';
+import Files from './components/Files';
+import { Readline } from 'xterm-readline';
+import { WebR } from '../webR/webr-main';
+import { CanvasMessage } from '../webR/webr-chan';
+import './App.css';
+
+const webR = new WebR({
+  RArgs: [],
+  REnv: {
+    R_HOME: '/usr/lib/R',
+    FONTCONFIG_PATH: '/etc/fonts',
+    R_ENABLE_JIT: '0',
+    COLORTERM: 'truecolor',
+  },
+});
+(globalThis as any).webR = webR;
+
+export interface TerminalInterface {
+  println: Readline['println'];
+  read: Readline['read'];
+  write: Readline['write'];
+}
+
+export interface FilesInterface {
+  refreshFilesystem: () => Promise<void>;
+  openFileInEditor: (name: string, path: string, readOnly: boolean) => Promise<void>;
+}
+
+const terminalInterface: TerminalInterface = {
```

Comment:
Generally: It seems strange to have "interface" in the name of an actual object?

```suggestion
const terminal: TerminalInterface = {
```